docs/data/base/components/modal/modal.md

```diff
@@ -0,0 +1,122 @@
+- Call the `onEnter` callback prop when the enter transition starts.
+- Call the `onExited` callback prop when the exit transition is completed.
+  These two callbacks allow the modal to unmount the child content when closed and fully transitioned.
+
+Modal has built-in support for [react-transition-group](https://github.com/reactjs/react-transition-group).
+
+{{"demo": "TransitionsModal.js", "defaultCodeOpen": false}}
+
+Alternatively, you can use [react-spring](https://github.com/pmndrs/react-spring).
+
+{{"demo": "SpringModal.js", "defaultCodeOpen": false}}
+
+## Performance
+
+The content of modal is unmounted when closed.
+If you need to make the content available to search engines or render expensive component trees inside your modal while optimizing for interaction responsiveness
+it might be a good idea to change this default behavior by enabling the `keepMounted` prop:
+
+```jsx
+<Modal keepMounted />
+```
+
+{{"demo": "KeepMountedModal.js", "defaultCodeOpen": false}}
+
+> You can use the `MuiModal-hidden` class to hide the modal when it is not opened.
+
+As with any performance optimization, this is not a silver bullet.
+Be sure to identify bottlenecks first, and then try out these optimization strategies.
+
+## Server-side modal
+
+React [doesn't support](https://github.com/facebook/react/issues/13097) the [`createPortal()`](https://reactjs.org/docs/portals.html) API on the server.
+In order to display the modal, you need to disable the portal feature with the `disablePortal` prop:
+
+{{"demo": "ServerModal.js", "defaultCodeOpen": false}}
+
+## Limitations
+
+### Focus trap
+
+The modal moves the focus back to the body of the component if the focus tries to escape it.
+
+This is done for accessibility purposes. However, it might create issues.
+In the event the users need to interact with another part of the page, e.g. with a chatbot window, you can disable the behavior:
+
+```jsx
+<Modal disableEnforceFocus />
+```
+
+## Accessibility
+
+(WAI-ARIA: https://www.w3.org/TR/wai-aria-practices/#dialog_modal)
+
+- Be sure to add `aria-labelledby="id..."`, referencing the modal title, to the `Modal`.
+  Additionally, you may give a description of your modal with the `aria-describedby="id..."` prop on the `Modal`.
+
+  ```jsx
+  <Modal aria-labelledby="modal-title" aria-describedby="modal-description">
+    <h2 id="modal-title">My Title</h2>
+    <p id="modal-description">My Description</p>
+  </Modal>
+  ```
+
+- The [WAI-ARIA authoring practices](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html) can help you set the initial focus on the most relevant element, based on your modal content.
+- Keep in mind that a "modal window" overlays on either the primary window or another modal window. Windows under a modal are **inert**. That is, users cannot interact with content outside an active modal window. This might create [conflicting behaviors](#focus-trap).
```

Comment:
I think this deserves to be called out as a cautionary note. Also edited for clarity.

```suggestion
> ⚠ **Note:** a modal window can sit on top of either the parent application, or another modal window. _All_ windows under the topmost modal are **inert**, meaning the user cannot interact with them. This can lead to [conflicting behaviors](#focus-trap).
```